wd_helper.au3

```diff
@@ -857,16 +859,26 @@ Func __WD_FrameList_Internal($sSession, $sLevel, $sFrameAttributes, $_WD_DEBUG_C
 				$sFrameAttributes = _WD_ExecuteScript($sSession, "return document.querySelectorAll('iframe')[" & $iFrame & "].outerHTML;", Default, Default, $_WD_JSON_Value)
 				$iErr = @error
 				If @error Then
-					$sMessage = 'Error occured on "' & $sLevel & '" level when trying to check atributes child frames #' & $iFrame
+					$sMessage = 'Error occured on "' & $sLevel & '" level when trying to check attributes child frames #' & $iFrame
 				Else
-					$sFrameAttributes = StringRegExpReplace($sFrameAttributes, '\R', '')
-					$vResult &= __WD_FrameList_Internal($sSession, $sLevel & '/' & $iFrame, $sFrameAttributes, $_WD_DEBUG_Calling)
+					Local $sScript = _
+							"var element = document.querySelectorAll('iframe')[" & $iFrame & "];" & _
+							"var style = window.getComputedStyle(element);" & _
+							"return (style.display === 'none');"
+					$bIsHidden = _WD_ExecuteScript($sSession, $sScript, Default, Default, $_WD_JSON_Value)
```

Comment:
Have you tested this? AFAICS, this won't be accurate when a parent container is hidden and not the actual frame.

Reply:
If parent container is hidden, do you expect that actual fram will be visible?

Reply:
Take a look on 
https://user-images.githubusercontent.com/11089482/186519722-2d7d707a-e1d4-49a4-af2b-92a05e6c4595.png

Is this the exact case you are refering to ?

Reply:
Yes, that is one instance. But you can't just check the parent element if that's what you are thinking.

Reply:
> Take a look on https://user-images.githubusercontent.com/11089482/186519722-2d7d707a-e1d4-49a4-af2b-92a05e6c4595.png
> 
> Is this the exact case you are refering to ?

IMHO this case should show 'IsHidden' status as true.

Reply:
> But you can't just check the parent element if that's what you are thinking.

No this is not my idea.

Reply:
> IMHO this case should show 'IsHidden' status as true.

Correct... I didn't mean to imply otherwise

> No this is not my idea.

ok

Reply:
So I do not understand.

Reply:
> this won't be accurate when a parent container is hidden and not the actual frame.

What you are refering here ?

Reply:
>>   this won't be accurate when a parent container is hidden and not the actual frame.

> What you are refering here ?

In this case, `a parent container` refers to the parent element *and* any of its ancestral parent elements. This is why I said "But you can't just check the parent element if that's what you are thinking" because simply checking the parentElement's visibility isn't sufficient to ensure that the frame is visible.

Reply:
You may want to test the `Displayed` functionality of _WD_ElementAction to see if it works correctly for this.

Reply:
Do you mean that we should check follow this: https://www.w3.org/TR/webdriver/#element-displayedness
`The visibility of an [element](https://www.w3.org/TR/webdriver/#dfn-elements) is guided by what is perceptually visible to the human eye. In this context, an [element](https://www.w3.org/TR/webdriver/#dfn-elements)’s displayedness does not relate to the [visibility](https://www.w3.org/TR/webdriver/#dfn-visibility) or [display](https://www.w3.org/TR/webdriver/#dfn-display) style properties.`

and use `_WD_ElementAction` with `DISPLAYED` instead checking it via `HTML DOM` ?

Reply:
Does this mean that element that is visible in `HTML DOM` but is not visible to the human eyes because it is on the 3rd page and to make it visible current view should be scroled down.... is treated differently?

differently... when I use 

- `HTML DOM` which should say is not hidden
- WEBDRIVER which should say is not visible to human eye, as this is far, far away on the bottom of the page.

?

Reply:
> > > this won't be accurate when a parent container is hidden and not the actual frame.
> 
> > What you are refering here ?
> 
> In this case, `a parent container` refers to the parent element _and_ any of its ancestral parent elements. This is why I said "But you can't just check the parent element if that's what you are thinking" because simply checking the parentElement's visibility isn't sufficient to ensure that the frame is visible.

On the follwing screenshot frame `#19  null/13/0` is hidden thus all sub frames `null/13/0/.......`  in this case `#20...#27` are also hidden and only `WebDriver test` shows  IsHiddenWebDriver=True but `JS test` shows IsHiddenJS=False thus ... `IsHiddenJS` shows invalid values.

This is what you was thinking about ?

![image](https://user-images.githubusercontent.com/11089482/221621770-100926fc-88ad-42ad-9e44-fcd7b771eda1.png)

Reply:
If so ... What I should do ? to leave only `IsHiddenWebDriver` feature and remove `IsHiddenJS` feature ?

Reply:
I suggest removing `IsHiddenJS` if it isn't accurate.

Reply:
Take a look again: [hidden status only from WebDriver - not JS](https://github.com/Danp2/au3WebDriver/pull/393/commits/bf60fce8e067e16a25b01ab6af054fe9ef433ce8)

Reply:
Can we cosider this part as resolved ?